components/supervisor/frontend/src/shared/frontend-dashboard-service.ts

```diff
@@ -46,6 +53,50 @@ export class FrontendDashboardServiceClient implements IDEFrontendDashboardServi
         return this.initPromise;
     }
 
+    decrypt(str: string): string {
+        if (!this.credentialsToken) {
+            throw new Error("no credentials token available");
+        }
+        const obj = JSON.parse(str);
+        if (!isSerializedEncryptedData(obj)) {
+            throw new Error("incorrect encrypted data");
+        }
+        const data = {
+            ...obj,
+            iv: Buffer.from(obj.iv, "base64"),
+            tag: Buffer.from(obj.tag, "base64"),
+        };
+        const decipher = crypto.createDecipheriv("aes-256-gcm", this.credentialsToken, data.iv);
```

Comment:
do we have a preference for the encryption algorithm? cc @geropl @akosyakov @securitymirco